clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -473,10 +473,20 @@ public StoreIngestionTask(
     this.chunkAssembler = new ChunkAssembler(storeName);
     this.cacheBackend = cacheBackend;
 
-    if (recordTransformerFunction != null) {
-      DaVinciRecordTransformer clientRecordTransformer = recordTransformerFunction.apply(versionNumber);
+    if (recordTransformerConfig != null && recordTransformerConfig.getRecordTransformerFunction() != null) {
+      Schema keySchema = schemaRepository.getKeySchema(storeName).getSchema();
+      // ToDo: Fetch latest value schema
```

Comment:
The latest can be fetched via `schemaRepository.getSupersetOrLatestValueSchema(storeName).getSchema()`